node/src/main/scala/com/wavesplatform/database/Caches.scala

```diff
@@ -442,7 +443,8 @@ abstract class Caches extends Blockchain, Storage {
     for (leaseId <- snapshot.cancelledLeases.keys) stateHash.addLeaseStatus(leaseId, isActive = false)
     for ((assetId, sponsorship) <- snapshot.sponsorships) stateHash.addSponsorship(assetId, sponsorship.minFee)
     for ((alias, address) <- snapshot.aliases) stateHash.addAlias(address, alias.name)
-    snapshot.nextCommittedGenerators.foreach(stateHash.addNextCommittedGenerator)
+    if (deterministicFinalityActivated)
+      snapshot.nextCommittedGenerators.foreach(stateHash.addNextCommittedGenerator)
```

Comment:
This is empty if the feature is not activated, because `CommitToGeneration` transaction is not allowed. No need for additional checks.